OpenSim/Simulation/Test/testInverseKinematicsSolver.cpp

```diff
@@ -91,6 +99,71 @@ int main()
 //=============================================================================
 // Test Cases
 //=============================================================================
+void testMarkersReference()
+{
+    vector<std::string> labels{ "A", "B", "C", "D", "E", "F" };
+    size_t nc = labels.size();
+    size_t nr = 5;
+
+    TimeSeriesTable_<SimTK::Vec3> markerData;
+    markerData.setColumnLabels(labels);
+    for (size_t r{0}; r < nr; ++r) {
+    SimTK::RowVector_<SimTK::Vec3> row{ int(nc), SimTK::Vec3(0) };
+    markerData.appendRow(0.1*r, row);
+    }
+
+    vector<int> order = { 3, 5, 1, 4, 0, 2};
+    Set<MarkerWeight> markerWeights;
+    for (size_t m{0}; m < nc; ++m)
+    markerWeights.adoptAndAppend(
+        new MarkerWeight(labels[order[m]], double(order[m])) );
```

Comment:
`order` needs to be at least as long as `labels`. Perhaps L115 should appear immediately after L104 so their relationship is clear.